### PR TITLE
chore: Add source code formatting

### PR DESCRIPTION
**Pull request by kaanyagci**
## Description

Closes [FL-1244](https://linear.app/owkin/issue/FL-1244/add-code-formatting-for-orchestrator)

### Changed

- `go` GitHub Action workflow updated checking code format added
- `Makefile` updated `format` target added and `lint` target updated to check the formatting  
- Source code formatted with `make format` target

## Checklist

- [ ] [changelog](../CHANGELOG.md) was updated with notable changes
- [ ] documentation was updated

### CODE REVIEW

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

File: docker/orchestrator-chaincode-init/dependencies.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [
   {
     "name": "curl",
-    "version": "8.3.0-r0"
+    "version": "8.4.0-r0"
```
Comment (oleobal) [inline docker/orchestrator-chaincode-init/dependencies.json:4]: This is addressed by #309, I think it's better if we avoid mixing concerns if possible

File: lib/asset/extended_addressable.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package asset
```
[the remark this answers is not in the document]
Comment (kaanyagci) [reply lib/asset/extended_addressable.go:1]: Sorry my bad, this file is committed accidentally I'll remove the file from PR

File: docker/orchestrator-chaincode-init/dependencies.json (hunk 1 above)
Comment (kaanyagci) [reply docker/orchestrator-chaincode-init/dependencies.json:4]: Yes indeed, I just saw the pull request, will drop that commit

**Review by oleobal — APPROVED** [review]
thanks!

**Review by SdgJlbl — COMMENTED** [review]
Nice addition 🚀 
Could we use something like `precommit` to have the formatter run everytime we commit? (yes I'm terribly lazy ^^)

**Comment by kaanyagci** [comment]
> Nice addition 🚀 Could we use something like `precommit` to have the formatter run everytime we commit? (yes I'm terribly lazy ^^)

Thank you very much! Apologies, I'm not that familiar with `precommit`, with a quick Google search I found there are [several community hooks for Go](https://pre-commit.com/hooks.html).  However, I think we can use the following script  as `pre-commit` hook in `.git/hooks/pre-commit`:

```bash
#!/bin/sh

make format

# Check for changes in the git index (i.e., changes that are staged)
git diff --cached --exit-code
```

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Review by SdgJlbl — APPROVED** [review]

**Pull request merged**